tez-api/src/main/java/org/apache/tez/runtime/api/ProcessorContext.java

```diff
@@ -96,27 +97,26 @@ default void setProgress(float progress) {
    * call. If invoking this method multiple times, it's recommended to remove
    * previously completed Inputs from the invocation list.
    *
-   * @param inputs
-   *          the list of Inputs to monitor
-   * @param timeoutMillis
-   *          timeout to return in milliseconds. If this value is negative,
-   *          this function will wait forever until all inputs get ready
-   *          or interrupted.
-   * @return the Input which is ready for consumption. return null when timeout occurs.
-   * @throws InterruptedException
+   * @param inputs the list of Inputs to monitor
+   * @param time the maximum time to wait
+   * @param unit the time unit of the time argument
+   * @return the {@code Input} which is ready for consumption or null if timeout
+   *         occurs
+   * @throws InterruptedException if the current thread is interrupted (and
+   *           interruption of thread suspension is supported)
    */
-  public Input waitForAnyInputReady(Collection<Input> inputs, long timeoutMillis) throws InterruptedException;
+  public Input waitForAnyInputReady(Collection<Input> inputs, long time, TimeUnit unit) throws InterruptedException;
```

Comment:
what about having a backward compatibility default interface method? 
this simply breaks the API which is used by hive upstream components (tez-api module),
https://github.com/apache/hive/blob/master/ql/src/java/org/apache/hadoop/hive/ql/exec/tez/HashTableLoader.java#L198
however this uses the single collection parameter version, but anyway, it's still easy to have a backward comp version of "inputs,  timeoutMillis", it can even be @Deprecated

Reply:
Hey @abstractdog,

Thanks for the input.  I'm obviously new to Tez, so I'm learning the process.  Since Tez isn't even 1.0 yet, I wasn't too worried about backwards compatibility, especially a pretty trivial one like this.  I would caution being too strict pre-1.0 and having a thousand deprecated methods, but if you feel strongly on this, I am happy to put in some backwards compatibility.